src/objectives/elbo/advi.jl

```diff
@@ -0,0 +1,127 @@
+
+"""
+    ADVI(n_samples; kwargs...)
+
+Automatic differentiation variational inference (ADVI; Kucukelbir *et al.* 2017) objective.
+This computes the evidence lower-bound (ELBO) through the ADVI formulation:
+```math
+\\begin{aligned}
+\\mathrm{ADVI}\\left(\\lambda\\right)
+&\\triangleq
+\\mathbb{E}_{\\eta \\sim q_{\\lambda}}\\left[
+  \\log \\pi\\left( \\phi^{-1}\\left( \\eta \\right) \\right)
+  +
+  \\log \\lvert J_{\\phi^{-1}}\\left(\\eta\\right) \\rvert
+\\right]
++ \\mathbb{H}\\left(q_{\\lambda}\\right),
+\\end{aligned}
+```
+where ``\\phi^{-1}`` is an "inverse bijector."
+
+# Arguments
+- `n_samples`: Number of Monte Carlo samples used to estimate the ELBO. (Type `<: Int`.)
```

Comment:
```suggestion
- `n_samples::Int`: Number of Monte Carlo samples used to estimate the ELBO.
```

Reply:
FYI `Int` is a concrete type and so `T <: Int` is only true if `T === Int` (`Int` is just an alias for the default integer representation on the device architecture, e.g. `Int64`)

Reply:
Oh, that's right.